### DISCUSSION TITLE
Database connection failed

### DISCUSSION THREAD

**Discussion by yellow-starburst** [idnahacks/GoodHound#61] [Q&A]
Despite removing every possible iteration of user-failure - it appears that I keep getting "database connection failure"

**Comment by idnahacks** [comment]
What command are you running and is your neo4j database running?

**Comment by yellow-starburst** [comment]
I tried the following connection options- I deleted the “-f csv…”
1. python3 goodhound.py -p password 
2. Change neo4j password to default neo4j 
3. python3 goodhound.py
4. Then I tried the full cmd 
5. python3 goodhound.py -p neo4j -u neo4j -s bolt://address
6. All attempts failed

**Comment by idnahacks** [comment]
Is your neo4j service running? All of those options should work ok. Is your neo4j server on the localhost?

**Comment by idnahacks** [comment]
I'm just installing it in Debian myself now in case there's a cross OS issue (I have tried previously without issues).

Update:
I can confirm that this has worked just fine in a fresh Windows and Debian install. Debian used pipenv to eliminate any library dependency issues. I'd definitely double check that you can connect to the neo4j web gui to confirm that the service is indeed running.

**Comment by yellow-starburst** [comment]
Well service is running and yeah it's localhost
let me see if I can try a different host/OS :)

**Comment by idnahacks** [comment]
Very strange. I've tried 3 separate installs here now with different OSes. All fine just running:
`python goodhound.py`

It does sound like a local issue, but please keep me posted on how you get on. If it's a bug at my end I'm committed to resolving it :)

**Comment by idnahacks** [comment]
Did you ever manage to find the cause of this @yellow-starburst ?
